src/reconstruction/CoreModel.jl

```diff
@@ -200,6 +200,60 @@ function verify_consistency(
     return (new_reactions, new_metabolites)
 end
 
+"""
+    remove_metabolites(model::CoreModel, metabolites)
+
+Removes a set of `metabolites` from the `model` of type `CoreModel` and returns
+a new `CoreModel` without those metabolites. Here, `metabolites` can be either a
+string, a vector of strings, an index or a vector of indices. Also removes any
+reactions that have no associated metabolites after the metabolites have been
+removed.
+
+# Example
+```
+model = load_model(CoreModel, "e_coli_core.json")
+
+m1 = remove_metabolites(model, ["glc__D_e", "for_c"])
+m2 = remove_metabolites(model, "glc__D_e")
+m3 = remove_metabolites(model, indexin(["glc__D_e", "for_c"], metabolites(model)))
+m4 = remove_metabolites(model, first(indexin(["glc__D_e"], metabolites(model))))
+```
+"""
+function remove_metabolites(model::CoreModel, mets)
```

Comment:
this is technically a variant (except for argument order), we'd also need to have a ! variant which would be more appropriate in this case.

Reply:
(for variant, you'd need to uncurry the arguments so that it can be used as `remove_metabolites("met123")(model)` )

Reply:
also, properly naming `remove_metabolites` vs `remove_metabolite` can help a lot with untangling ambiguous cases

Reply:
This will be addressed in a feature PR though, see issue #281